src/option_menu.c

```diff
@@ -0,0 +1,929 @@
+
+
+//This file's functions
+static void sub_808835C(void);
+static void sub_8088374(void);
+static void sub_8088430(void);
+static void sub_8088444(void);
+static void sub_8088454(void);
+static void sub_8088514(void);
+static void sub_8088530(void);
+static void sub_8088680(void);
+static void sub_80886D4(void);
+static bool8 sub_80886F0(void);
+static void sub_8088780(u8 taskId);
+static u8 sub_80888C0(void);
+static void sub_80889A8(u8 selection);
+static void sub_8088B00(u8 taskId);
+static void sub_8088BD0(void);
+static void sub_8088C0C(void);
+static void sub_8088D8C(void);
+static void sub_8088DE0(u16 selection);
+
+struct OptionsMenu
+{
+    /*0x00*/ u16 option[MENUITEM_COUNT];    //0,2,4,6,8,a,c
+    /*0x0E*/ u16 unkE;
+    /*0x10*/ u8 state3;
+    /*0x11*/ u8 state;
+    /*0x12*/ u8 state2;
+    /*0x13*/ u8 unk13;
+};
+
+EWRAM_DATA struct OptionsMenu *sOptionsMenuPtr = {0};
+
+
+
+//CB2_InitOptionsMenu
+static void sub_808835C(void)
+{
+    RunTasks();
+    AnimateSprites();
+    BuildOamBuffer();
+    UpdatePaletteFade();
+}
+
+
+//VBlankCB_OptionsMenu
+static void sub_8088374(void)
+{
+    LoadOam();
+    ProcessSpriteCopyRequests();
+    TransferPlttBuffer();
+}
+
+
+//sub_8088388
+void CB2_OptionsMenuFromStartMenu(void)
+{
+    u8 i;
+    
+    if (gMain.savedCallback == NULL)
+        gMain.savedCallback = CB2_ReturnToStartMenu;
+    sOptionsMenuPtr = AllocZeroed(sizeof(struct OptionsMenu));
+    sOptionsMenuPtr->state3 = 0;
+    sOptionsMenuPtr->state2 = 0;
+    sOptionsMenuPtr->state = 0;
+    sOptionsMenuPtr->unkE = 0;
+    sOptionsMenuPtr->option[MENUITEM_TEXTSPEED] = gSaveBlock2Ptr->optionsTextSpeed;
+    sOptionsMenuPtr->option[MENUITEM_BATTLESCENE] = gSaveBlock2Ptr->optionsBattleSceneOff;
+    sOptionsMenuPtr->option[MENUITEM_BATTLESTYLE] = gSaveBlock2Ptr->optionsBattleStyle;
+    sOptionsMenuPtr->option[MENUITEM_SOUND] = gSaveBlock2Ptr->optionsSound;
+    sOptionsMenuPtr->option[MENUITEM_BUTTONMODE] = gSaveBlock2Ptr->optionsButtonMode;
+    sOptionsMenuPtr->option[MENUITEM_FRAMETYPE] = gSaveBlock2Ptr->optionsWindowFrameType;
+    
+    for (i = 0; i < MENUITEM_COUNT-1; i++)
+    {
+        if (sOptionsMenuPtr->option[i] > (sOptionsMenuItemCounts[i])-1)
+            sOptionsMenuPtr->option[i] = 0;
+    }
+    HelpSystem_SetSomeVariable2(0xD);
+    SetMainCallback2(sub_8088454);
+}
+
+
+//sub_8088430
+static void sub_8088430(void)
+{
+    SetVBlankCallback(0);
+    SetHBlankCallback(0);
+}
+
+
+//sub_8088444
+static void sub_8088444(void)
+{
+    SetVBlankCallback(sub_8088374);
+}
+
+//sub_8088454
+static void sub_8088454(void)
+{
+    u8 i, state;
+    state = sOptionsMenuPtr->state;
+    switch (state)
+    {
+    case 0:
+        sub_8088430();
+        break;
+    case 1:
+        sub_8088530();
+        break;
+    case 2:
+        sub_80886D4();
+        break;
+    case 3:
+        if (sub_80886F0() != TRUE)    //silly matching quirk
```

Comment:
This is actually common through the project, so the comment should be removed here